include/singa/neuralnet/input_layer.h

```diff
@@ -57,6 +62,8 @@ class StoreInputLayer : virtual public InputLayer {
   int batchsize_ = 1;
   int random_skip_ = 0;
   io::Store* store_ = nullptr;
+  vector<std::string> buf_keys_, buf_vals_;
+  std::deque<std::thread> threads_;  // prefetching thread
```

Comment:
may not need a deque? there is only one prefetching thread per store input layer?